client/lib/sites-list/list.js

```diff
@@ -167,28 +167,44 @@ SitesList.prototype.createSiteObject = function( site ) {
 		return Site( site );
 	}
 };
+
 /**
  * Marks collisions between .com sites and Jetpack sites that have the same URL
  * Add the hasConflict attribute to .com sites that collide with Jetpack sites.
  *
  * @api private
  *
+ * @param {Array} sites list of stored sites
  */
 SitesList.prototype.markCollisions = function( sites ) {
-	sites.forEach( function( site, index, collisions ) {
-		var hasCollision;
+	const jetpackUrls = new Map();
+
+	const addJetpackSiteUrl = ( { ID, URL } ) => {
+		const baseUrl = withoutHttp( URL );
+		let siteSet = jetpackUrls.get( baseUrl );
+
+		if ( undefined === siteSet ) {
+			siteSet = new Set();
```

Comment:
Could we not just store the site url? Why do we need the ID? Is there a reason to think that they might not be unique?

Reply:
See my comment below - I wrote this from a perspective of ignorance, so I have tried as hard as I can to preserve existing behavior and I think the existing behavior here depends on knowing what is stored in a URL. If we never get more than one and rely on that assumption then I'm all for simplifying.